include/boost/json/value.hpp

```diff
@@ -2832,6 +2832,65 @@ struct key_value_pair
 
 //----------------------------------------------------------
 
+#ifdef BOOST_JSON_DOCS
```

Comment:
When we use `BOOST_JSON_DOCS` we need to strive to make the conditionally compiled section as small as possible. This exposes more of the doc material to the compiler and helps catch errors. For example we would prefer:
```
#ifdef BOOST_JSON_DOCS
__see_below__
#else
ugly_type
#endif
great_function( complicated_args... );
```

over

```
#ifdef BOOST_JSON_DOCS
__see_below__
great_function( complicated_args... );
#else
ugly_type
great_function( complicated_args... );
#endif
```

So I think, we should try to apply this to the code here (not sure what the best way is).